source/sessions/driver-sessions.rst

```diff
@@ -992,9 +992,9 @@ to the pool.
 
 8. Client-side cursor that exhausts the results after a ``getMore`` immediately returns the implicit session
 to the pool.
-    * Insert four documents into a collection
-    * Execute a find operation on the collection with batch size of 2
-    * Iterate past the first three documents, forcing the final ``getMore`` operation
+    * Insert five documents into a collection
+    * Execute a find operation on the collection with batch size of 3
+    * Iterate past the first four documents, forcing the final ``getMore`` operation
```

Comment:
How is this any different? You've just added +1 to each number.

Reply:
With 4 documents and batchSize=2, the server returns two getMores. The first getMore contains documents three and four and includes a non-zero cursorId. The second getMore contains zero documents and a cursorId of zero.

With 5 documents and batchSize=3, the server returns a single getMore. The getMore contains documents four and five and includes a cursorId of zero.

Reply:
I see... the server does something a bit unexpected when the last batch is full?

Reply:
Yeah, if the last batch has exactly batchSize documents, the server returns it with a non-zero cursorId, followed by an empty batch with cursorId zero in response to the final getMore.